src/modules/service-catalog/ServiceCatalogItemPage.tsx

```diff
@@ -97,43 +32,65 @@ export function ServiceCatalogItemPage({
   brandId,
 }: ServiceCatalogItemPageProps) {
   const serviceCatalogItem = useServiceCatalogItem(serviceCatalogItemId);
-  const { requestFields, handleChange } = useItemFormFields(
+  const {
+    requestFields,
+    associatedLookupField,
+    setRequestFields,
+    handleChange,
+  } = useItemFormFields(serviceCatalogItem, baseLocale);
+  const { submitServiceItemRequest } = useServiceFormSubmit(
     serviceCatalogItem,
+    requestFields,
+    associatedLookupField,
     baseLocale
   );
-  const { t } = useTranslation();
+
+  const handleRequestSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const response = await submitServiceItemRequest();
+    if (response && !response.ok) {
+      const errorData = await response.json();
+      if (errorData.error === "RecordInvalid") {
+        const invalidFieldErros = errorData.details.base;
+        const updatedFields = requestFields.map((field) => {
+          const errorField = invalidFieldErros.find(
+            (errorField: RequestError) => errorField.field_key === field.id
+          );
+          return errorField
+            ? { ...field, error: errorField.description }
+            : field;
+        });
+        setRequestFields(updatedFields);
+      }
+    } else if (response && response.ok) {
+      const data = await response?.json();
+      const redirectUrl = "/hc/requests/" + data.request.id;
+      window.location.href = redirectUrl;
```

Comment:
We also need to show a success notification on the request page, as shown in the designs. We can use the `addFlashNotification` function as we do in the answer bot modal: https://github.com/zendesk/copenhagen_theme/blob/a9a12f06f07d5919f36e0b33505d52a789a927d9/src/modules/new-request-form/answer-bot-modal/AnswerBotModal.tsx#L107

Reply:
Ok added. I'm not sure if we will need it also for error but I will need to ask for copy.

Reply:
Do we have an answer to this? IMO if the submission fails for other reasons we would need to show an error somewhere

Reply:
I've just go the copy so I added the correct one